src/main/java/be/seeseemelk/mockbukkit/MockPlayerList.java

```diff
@@ -62,10 +62,21 @@ public BanList getProfileBans()
 
 	public void addPlayer(@NotNull PlayerMock player)
 	{
+		if(isPlayerAlreadyOnServer(player))
+			throw new IllegalStateException("Player is already on server!");
+
 		onlinePlayers.add(player);
 		offlinePlayers.add(player);
 	}
 
+	public void removePlayer(@NotNull PlayerMock player) throws IllegalStateException
+	{
+		if(onlinePlayers.contains(player))
+			onlinePlayers.remove(player);
+		else
+			throw new IllegalStateException("Player is not connected!");
```

Comment:
This can be simplified, no need to perform two operations here, we get the lookup for free.

```suggestion
		if (!onlinePlayers.remove(player)) {
			throw new IllegalStateException("Player is not connected!");
		}
```